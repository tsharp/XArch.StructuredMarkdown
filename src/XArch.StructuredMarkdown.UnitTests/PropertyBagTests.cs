namespace XArch.StructuredMarkdown.UnitTests
{
    using Snapshooter.Xunit;

    using XArch.StructuredMarkdown.Serialization;

    public class PropertyBagTests
    {
        [Fact]
        public void CanParsePropertyBag()
        {
            File.ReadAllText("./test-data/metadata/raw.yml")
                .FromYaml<PropertyBag>()
                .MatchSnapshot();
        }

        [Fact]
        public void CanParseSerialziePropertyBag()
        {
            PropertyBag bag = new PropertyBag
            {
                { "title", "Test Document" },
                { "author", "John Doe" },
                { "tags", new List<string> { "test", "example" } }
            };

            bag.ToYaml()
                .MatchSnapshot();
        }
    }
}