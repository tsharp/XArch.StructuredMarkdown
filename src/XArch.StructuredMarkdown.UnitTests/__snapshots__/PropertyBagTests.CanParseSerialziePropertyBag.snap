title: Test Document
author: John Doe
tags:
- test
- example
